pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -182,6 +182,7 @@ public void onUpdate(String path, Set<String> data, Stat stat) {
                 if (log.isDebugEnabled()) {
                     log.debug("Update Received for path {}", path);
                 }
+                reapDeadBrokerPreallocations(data);
```

Comment:
quick question: we are triggering `reapDeadBrokerPreallocations` on children-change, should we also trigger it when [broker-data changes](https://github.com/yahoo/pulsar/blob/master/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java#L512)?